### PR TITLE
Add support for RPL_UMODEIS

### PR DESCRIPTION
**Pull request by progval**
https://modern.ircdocs.horse/#rplumodeis-221

### CODE REVIEW

File: src/sirc/sirc_event_hdr.c [hunk 1]
```diff
@@ -72,11 +72,15 @@ void sirc_event_hdr(SircSession *sirc, SircMessage *imsg){
...
-                events->numeric(sirc, num, origin, params, imsg->nparam);
+                break;
+            case SIRC_RFC_RPL_UMODEIS:
+                 /* User mode changed */
+                 g_return_if_fail(events->umode);
+                 events->umode(sirc, imsg->cmd, origin, params, imsg->nparam);
+                 break;
         }
+        g_return_if_fail(events->numeric);
+        events->numeric(sirc, num, origin, params, imsg->nparam);
```
Comment (SilverRainZ) [inline src/sirc/sirc_event_hdr.c:83]: `event->numeric` will be called after `break` out from every case, I think it is unexpected.

File: src/core/app_irc_event.c [hunk 2]
```diff
@@ -1836,6 +1836,10 @@ static void irc_event_numeric(SircSession *sirc, int event,
                         _("URL of %1$s is: %2$s"), chan, msg);
                 break;
             }
+        case SIRC_RFC_RPL_UMODEIS:
+            {
+                break;
+            }
```
Comment (SilverRainZ) [inline src/core/app_irc_event.c:1842]: This change is not needed if we put the call of `events->numeric` in the default case.

**Review by SilverRainZ — CHANGES_REQUESTED** [review]

File: src/sirc/sirc_event_hdr.c (hunk 1 above)
Comment (progval) [reply src/sirc/sirc_event_hdr.c:83]: I thought it could be confusing if `irc_event_numeric` was called for all numerics except a few

Comment (SilverRainZ) [reply src/sirc/sirc_event_hdr.c:83]: It is a fallback of "non-special" numeric event, it maybe a little confusing, but call it for all numerics is incorrect for now.

Comment (SilverRainZ) [reply src/sirc/sirc_event_hdr.c:83]: You should not move `event->numeric` out of default case of switch clause.

**Review by SilverRainZ — CHANGES_REQUESTED** [review]

**Review by SilverRainZ — APPROVED** [review]

**Pull request merged**